src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/ModelBindingContext.cs

```diff
@@ -289,12 +310,60 @@ public IValueProvider ValueProvider
         /// Gets or sets a predicate which will be evaluated for each property to determine if the property
         /// is eligible for model binding.
         /// </summary>
-        public Func<ModelBindingContext, string, bool> PropertyFilter { get; set; }
+        public Func<IModelBindingContext, string, bool> PropertyFilter
+        {
+            get { return _state.PropertyFilter; }
+            set { _state.PropertyFilter = value; }
+        }
 
         /// <summary>
         /// Gets or sets the <see cref="ValidationStateDictionary"/>. Used for tracking validation state to
         /// customize validation behavior for a model object.
         /// </summary>
-        public ValidationStateDictionary ValidationState { get; set; }
+        public ValidationStateDictionary ValidationState
+        {
+            get { return _state.ValidationState; }
+            set { _state.ValidationState = value; }
+        }
+
+        public ModelBindingResult? Result
```

Comment:
`/// <inheritdoc />` and, if you like, can slim most of the other documentation down to this.